packages/web-components/fast-components-msft/src/badge/index.ts

```diff
@@ -0,0 +1,26 @@
+import { attr, customElement } from "@microsoft/fast-element";
+import { Badge, BadgeTemplate as template } from "@microsoft/fast-components";
+import { BadgeStyles as styles } from "./badge.styles";
+
+export type BadgeFill = "accent" | "lightweight" | "neutral";
+
+@customElement({
+    name: "msft-badge",
+    template,
+    styles,
+})
+export class MSFTBadge extends Badge {
+    @attr
+    public fill: BadgeFill = "lightweight";
+    private fillChanged(oldValue: BadgeFill, newValue: BadgeFill): void {
+        if (oldValue !== newValue) {
+            this.classList.add(newValue);
+            this.classList.remove(oldValue);
+        }
+    }
+
+    // Can we remove this and circular as attrs if we don't want to leverage them?
+    // Can I set this internally only in this implementation?
+    @attr({ mode: "fromView" })
+    public color: string = "value";
+}
```

Comment:
@microsoft/fast-dna-collaborators my thoughts on the implementation of Badge here was to tighten up the implementation to better align with the MSFT badge requirements. Ideally I'd like to not provide the color value here  and just have things work as expected. Since we cannot remove or delete attributes from the classes here, I'm trying to consider how to manage the mappings. 

It seems like having defaults for both `fill` and `color` make sense; while not illustrated above we could also ensure the extensibility by also supporting a string:
```ts
export type BadgeFill = "accent" | "lightweight" | "neutral" | string;
```
I'm not sure what to do with the color attribute in those scenarios though if we elect to handle them in the stylesheet.

Reply:
The `fast-components` api would remain the same?

Reply:
I think this is part of the question...since we can't pull off existing attrs, extending from the Badge would expose all of those values - so yes, the API would stay the same. I think providing some default "named" configurations is helpful here, but wondering what our language/naming should align to. Recipes?

Reply:
I think the naming seems right to me. It aligns with what we do elsewhere.

Reply:
The one thing about our react `badge` is that the `filled` version was yellow and not aligned with the accent color. So might want @bheston to chime in here.

Reply:
Yeah, I'd like some thoughts on the defaults - we can still keep things extensible. My concern with the yellow is that it won't respond to dark mode unless we leverage the provider which seems heavy-handed for a default value.

Bigger question in my mind is what to do with the color value (if anything)...do we provide it but not include a default value b/c the fill defaults will be recipe associated?

Reply:
One option might be to move `color` so that it's defined on `FASTBadge` rather than `Badge`.

Reply:
@EisenbergEffect there would be changes to the template as well I think - those values are set in a nested div as they are setting `style` attributes. I wonder if the deltas here are worth different API's or if we should keep consistent for the time being.

Reply:
Ah, I see. Well, then it becomes a matter of what is worse, having some properties that don't apply vs. writing a whole new component.